### ISSUE TITLE
[Feature] Trick to tell AMD and Nvidia drivers to use the most powerful GPU instead of a lower-performance (such as integrated) GPU

### ISSUE THREAD

**Issue by shellka** [nwjs/nw.js#5732]
Hello.
It would be great if we will have options to enable powerful GPU
https://gist.github.com/statico/6809850727c708f08458

**Comment by StrategJuvty** [comment]
yup, I need it too

**Comment by rogerwang** [comment]
This should be enabled in 0.21.0 rc versions with Chromium 57:
https://skia.googlesource.com/skia/+/chrome/m57/tools/gpu/GrContextFactory.cpp#33
https://skia.googlesource.com/skia/+/chrome/m57/BUILD.gn#30

Please try with our latest 0.21.0 nightly build: http://dl.nwjs.io/live-build/02-28-2017/nw21-0a37a7f-6a70483-e6f163e-1c48b0a/v0.21.0/

**Comment by StrategJuvty** [comment]
@rogerwang This build doesn't work. GPU Nvidia Optimus started on integrated driver as usual. And a can't see any export of NvOptimusEnablement in binaries. Could you check it please?

**Comment by rogerwang** [comment]
Will do. How did you test it besides checking the symbols?

**Comment by StrategJuvty** [comment]
@rogerwang By text search "NvOptimusEnablement" in files. You can check any hardware-accelerated game exe file for that string. This string is always present for trick the drivers.

**Comment by shellka** [comment]
To complete test you need to have laptop with built-in and nvidia GPU - you should see that by default it launch with built-in GPU, but if the app has this constant - it should launch on nvidia (amd).

**Comment by TheRealDannyyy** [comment]
I would like to request an update about the current progress of this feature.
Is it already available in nightly builds and does it run as intended?

**Comment by rogerwang** [comment]
No. It's still not fixed. The code I referred to previously was for skia testing tools. Will keep this issue updated.

**Comment by Danwood87** [comment]
How's this option enabled in the latest version?

**Comment by TheRealDannyyy** [comment]
@rogerwang I would appreciate a quick update for this one. Is it perhaps already implemented but requires to be enabled manually?

**Comment by rogerwang** [comment]
We didn't fix it. Will see it soon.

**Comment by rogerwang** [comment]
btw, could you please check the status in Chrome? Currently we don't have the hardware to check it.

**Comment by lachsen** [comment]
Hey!
This issue is very old an still open.
Are there any news about this?

We're creating a game with NW.js and by default, it always uses integrated GPU on Windows (mostly on Laptops)

**Comment by rogerwang** [comment]
That just controls the default setting. In the system, a tool (such as Nvidia panel) is used to control which gpu is used by each program.

**Comment by lisliz** [comment]
Can you go ahead and fix this? 
Electron seems to have --force_high_performance_gpu to fix the issue, but I don't want to use Electron.
I'm a game developer. NW.js runs with integrated graphics by default and it's causing a lot of refunds. Players usually don't adjust Nvidia panel settings.

**Comment by lisliz** [comment]
I found that the --force_high_performance_gpu flag seems to make NW.js run with a discrete graphics card by default, but with the --in-process-gpu flag it will not work.

**Comment by alastajj** [comment]
@lisliz thank you, that makes laptop GPU work.

Is there a way to make it work with --in-process-gpu? 

Because Steam Overlay needs it, I think.
